source/core/parameters.cc

```diff
@@ -542,7 +519,46 @@ namespace Parameters
                         "false",
                         Patterns::Bool(),
                         "Non Newtonian flow");
+
+      prm.declare_entry("rheological model",
+                        "newtonian",
+                        Patterns::Selection("newtonian|power-law|carreau"),
+                        "Rheological model "
+                        "Choices are <newtonian|power-law|carreau>.");
+
       non_newtonian_parameters.declare_parameters(prm);
+
+
+      prm.declare_entry("density model",
+                        "constant",
+                        Patterns::Selection("constant"),
+                        "Model used for the calculation of the density"
+                        "Choices are <constant>.");
+
+      prm.declare_entry("specific heat model",
+                        "constant",
+                        Patterns::Selection("constant|phase_change"),
+                        "Model used for the calculation of the specific heat"
+                        "Choices are <constant|phase_change>.");
+
+      phase_change_parameters.declare_parameters(prm);
+
+      prm.declare_entry(
+        "thermal conductivity model",
+        "constant",
+        Patterns::Selection("constant|linear"),
+        "Model used for the calculation of the thermal conductivity"
+        "Choices are <constant|linear>.");
+
+      prm.declare_entry("k_A0",
```

Comment:
we should think about a more explicit name in the .prm, for this parameters

Reply:
I agree!

Reply:
Abs-So-Lu-Tely. I Just did not know how to name them. Nobody uses it for now, but we can clearly revisit that. If you have any ideas, i'll gladly do it now

Reply:
For this PR you can merge to get going, I need to give some thoughts about it. @CaroleAnneDaunais let's brainstorm that together :)